src/config.c

```diff
@@ -2474,6 +2474,31 @@ static int updateJemallocBgThread(const char **err) {
     return 1;
 }
 
+static int updateActiveRehashing(const char **err) {
+    UNUSED(err);
+    if (!server.cluster_enabled) return 1;
+
+    if (server.activerehashing) {
+        /* Add the dictionaries which are undergoing rehashing to the `rehashing` list.
+        * It will be picked up by the incremental rehash logic in the serverCron. */
+        for (dbKeyType subdict = DB_MAIN; subdict <= DB_EXPIRES; subdict++) {
+            dict *d;
+            dbIterator *dbit = dbIteratorInit(&server.db[0], subdict);
+            while ((d = dbIteratorNextDict(dbit))) {
+                if (dictIsRehashing(d)) {
+                    listAddNodeTail(server.db->sub_dict[subdict].rehashing, d);
```

Comment:
i have a concern about this loop causing a short freeze when the config is changed.
although i suppose it can't be that bad, considering that until #12697 we have a similar loop in every INFO call.

still, considering that this config is mainly meant to avoid latency issues from long cron, maybe we can let redis keep track of the rehashed dicts even when disabled?
i.e. the config will reduce latency issues, but not the memory overhead.

i suppose it depends if users that disable this config do it only temporarily (when the time latency sensitive period, and re-enable it later), or do they keep it permanently disabled.

does anyone have any suggestion / opinion / info?

Reply:
Not adding the missed out dictionaries to the list could prolong the rehash for quite a long period of time. I believe it would be good to update the list (note: it skips the dictionaries which doesn't have any data).

Reply:
@judeng Do you have any insights on this?

Reply:
Now we always disable the config to avoid latency and cpu overload. IMO, temporary operation  is too late because rehashing doesn't last long.

Maybe we can separate rehashing lists from `incrementallyRehash` function, we could rotate the list and check if dicts' rehashing is completed, no matter `activerehashing` is disabled or enabled.

Reply:
> Maybe we can separate rehashing lists from `incrementallyRehash` function, we could rotate the list and check if dicts' rehashing is completed, no matter `activerehashing` is disabled or enabled.

i don't think that's needed, the changes in #12697 would still clean this list when the dict reshard ends (regardless of this config).

> Now we always disable the config to avoid latency and cpu overload. IMO, temporary operation is too late because rehashing doesn't last long.

i meant that if you have periods of low traffic, or low latency demands, it can be enabled for these periods.

> Not adding the missed out dictionaries to the list could prolong the rehash for quite a long period of time. I believe it would be good to update the list (note: it skips the dictionaries which doesn't have any data).

do you mean that it's a good idea to keep maintaining it even when reshard is disabled (small memory overhead), or you think the latency spike of re-computing it (current code) is ok?
maybe you can measure it?